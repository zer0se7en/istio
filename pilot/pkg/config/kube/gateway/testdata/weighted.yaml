apiVersion: networking.x-k8s.io/v1alpha1
kind: GatewayClass
metadata:
  name: istio
spec:
  controller: istio.io/gateway-controller
---
apiVersion: networking.x-k8s.io/v1alpha1
kind: Gateway
metadata:
  name: gateway
  namespace: istio-system
spec:
  gatewayClassName: istio
  listeners:
  - hostname: "*.domain.example"
    port: 80
    protocol: HTTP
    routes:
      namespaces:
        from: All
      kind: HTTPRoute
---
apiVersion: networking.x-k8s.io/v1alpha1
kind: HTTPRoute
metadata:
  name: http
  namespace: default
spec:
  hostnames: ["first.domain.example"]
  rules:
  - matches:
    - path:
        type: Prefix
        value: /get
    forwardTo:
      - serviceName: httpbin
        port: 80
        weight: 2
      - serviceName: httpbin-other
        port: 8080
        weight: 3
