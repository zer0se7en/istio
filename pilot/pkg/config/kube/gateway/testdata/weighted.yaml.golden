apiVersion: networking.istio.io/v1alpha3
kind: Gateway
metadata:
  annotations:
    internal.istio.io/gateway-service: istio-ingressgateway.istio-system.svc.domain.suffix
  creationTimestamp: null
  name: gateway-istio-autogenerated-k8s-gateway
  namespace: istio-system
spec:
  servers:
  - hosts:
    - '*.domain.example'
    port:
      name: 0-gateway-gateway-istio-system
      number: 80
      protocol: HTTP
  - hosts:
    - '*'
    port:
      name: 1-gateway-gateway-istio-system
      number: 34000
      protocol: TCP
---
apiVersion: networking.istio.io/v1alpha3
kind: VirtualService
metadata:
  creationTimestamp: null
  name: tcp-tcp-istio-autogenerated-k8s-gateway
  namespace: default
spec:
  gateways:
  - istio-system/gateway-istio-autogenerated-k8s-gateway
  hosts:
  - '*'
  tcp:
  - route:
    - destination:
        host: httpbin.default.svc.domain.suffix
        port:
          number: 9090
      weight: 33
    - destination:
        host: httpbin-alt.default.svc.domain.suffix
        port:
          number: 9090
      weight: 67
---
apiVersion: networking.istio.io/v1alpha3
kind: VirtualService
metadata:
  creationTimestamp: null
  name: http-istio-autogenerated-k8s-gateway
  namespace: default
spec:
  gateways:
  - istio-system/gateway-istio-autogenerated-k8s-gateway
  hosts:
  - first.domain.example
  http:
  - match:
    - uri:
        prefix: /get
    route:
    - destination:
        host: httpbin.default.svc.domain.suffix
        port:
          number: 80
      weight: 40
    - destination:
        host: httpbin-other.default.svc.domain.suffix
        port:
          number: 8080
      weight: 60
  - match:
    - uri:
        prefix: /weighted-100
    route:
    - destination:
        host: foo-svc.default.svc.domain.suffix
        port:
          number: 8000
      headers:
        request:
          add:
            foo: bar
---
